sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/commands.scala

```diff
@@ -102,7 +103,12 @@ private[sql] case class InsertIntoHadoopFsRelation(
       case (SaveMode.ErrorIfExists, true) =>
         throw new AnalysisException(s"path $qualifiedOutputPath already exists.")
       case (SaveMode.Overwrite, true) =>
-        fs.delete(qualifiedOutputPath, true)
+        Utils.tryOrIOException {
```

Comment:
Throw exception if failed to delete the files, otherwise it will case failure in the unit test, which expect exception raised.